src/main/typescript/rug/operations/Decorators.ts

```diff
@@ -27,137 +30,190 @@ function get_metadata(obj: any, key: string) {
 /**
  * Decorator for parameters. Adds to object properties
  */
-function Parameter(details: BaseParameter) {
+export function Parameter(details: BaseParameter) {
   return (target: any, propertyKey: string) => {
-    let params = get_metadata(target, "__parameters");
-    if (params == null) {
-      params = [];
-    }
-    const copy: any = { ...details };
-    copy.name = propertyKey;
-    copy.decorated = true;
-    params.push(copy);
-    set_metadata(target, "__parameters", params);
+    declareParameter(target, propertyKey, details);
   };
 }
+
+export function declareParameter(target: any, propertyKey: string, details: BaseParameter) {
```

Comment:
It's not a big deal. Four-space indentation seems to be the standard for TypeScript. It's what tsfmt uses. I'm surprised vscode is not using that by default. It seems to in other projects. Perhaps it detects what is being used in the file and just goes with that.